### ISSUE TITLE
Feature request: `noInlineConfig` flag

### ISSUE THREAD

**Issue by webbertakken** [oxc-project/oxc#15173]
Trying to keep some projects clean by not allowing inline disables in favour of proper configuration of fitting rules for the project. I'm liking it in my ESLint config and would like to apply it to Oxlint as well before migrating.

In ESLint the following configuration can be used to not allow inline comments:

```ts
  {
    linterOptions: {
      noInlineConfig: true, // ignores all /* eslint-disable */ etc.
      reportUnusedDisableDirectives: 'error',
    },
  }
```

For Oxlint the best we have (that I could find) is 

```jsonc
  // Base rules
  "rules": {
    "unicorn/no-abusive-eslint-disable": "error"
  },
```

We also have 

```bash
# report unused rules as warnings
oxlint --report-unused-disable-directives

# report unused as errors
oxlint --report-unused-disable-directives-severity=error
```

But there is currently no way to disallow inline config completely for a project.

It would be really nice if we could have `noInlineConfig` setting or rule for Oxlint.

## Considered alternatives

We could probably do this with a custom rule

**Comment by kwangure** [comment]
I'm wondering how to configure `--report-unused-disable-directives` and `--report-unused-disable-directives-severity=error` in `.oxlintrc.json` instead of CLI so that it works for VS Code as well. Is this supported?

**Comment by webbertakken** [comment]
@kwangure You can use "oxc.unusedDisableDirectives" in VS Code

See https://marketplace.visualstudio.com/items?itemName=oxc.oxc-vscode#:~:text=oxc.unusedDisableDirectives

**Comment by kwangure** [comment]
Just what I needed! I didn't think to check there. Maybe there should be a VS Code page on the OXC site with these.

Also, it feels like some of these VS Code settings/CLI args should also be options in `.oxlintrc.json` where they're applied to _both_ the LSP and via CLI. 

Anyway. This is sufficient for now. Thanks Webber!

**Comment by connorshea** [comment]
I like the idea of a webpage for the VSCode extension, you should open an issue about that

**Comment by connorshea** [comment]
> I'm wondering how to configure `--report-unused-disable-directives` and `--report-unused-disable-directives-severity=error` in `.oxlintrc.json` instead of CLI so that it works for VS Code as well. Is this supported?

Issue for doing this via the `.oxlintrc.json` is here, hopefully will be added in the near-ish future: https://github.com/oxc-project/oxc/issues/15972

**Comment by hybrist** [comment]
We were trying to use oxlint instead of ESLint for checks against security rules (e.g. https://github.com/google/safety-web) and we really don't want inline config to be able to disable those. Especially not in the age of AI agents who are eager to "get things done".
